### PR TITLE
[#82] Add trainee activity and progress read API for trainers

### PR DESCRIPTION
**Pull request by withelm**
## Summary
- add trainer-scoped read endpoints for trainee training history/details, exercise progress chart, elo chart, and main records history
- enforce trainer-to-trainee ownership boundary in application service before delegating reads to existing domain services
- extend trainer relationship integration tests with ownership isolation, invalid-id edge cases, and data integrity checks for new endpoints

## Verification
- dotnet test LgymApi.IntegrationTests/LgymApi.IntegrationTests.csproj --filter \"FullyQualifiedName~TrainerRelationshipTests\"
- dotnet test LgymApi.UnitTests/LgymApi.UnitTests.csproj --filter \"FullyQualifiedName~EndpointResponseTypeTests|FullyQualifiedName~MapperConfigurationTests\"

### CODE REVIEW

File: LgymApi.Api/Features/Trainer/Controllers/TrainerRelationshipController.cs [hunk 1]
```diff
@@ -73,6 +80,87 @@ public async Task<IActionResult> GetDashboardTrainees([FromQuery] TrainerDashboa
...
+        if (!Guid.TryParse(traineeId, out var parsedTraineeId))
+        {
+            throw AppException.BadRequest(Messages.UserIdRequired);
+        }
+
+        var trainer = HttpContext.GetCurrentUser();
+        var records = await _trainerRelationshipService.GetTraineeMainRecordsHistoryAsync(trainer!, parsedTraineeId);
+        var mappedRecords = _mapper.MapList<LgymApi.Domain.Entities.MainRecord, MainRecordResponseDto>(records);
+        return Ok(mappedRecords);
+    }
```
[the remark this answers is not in the document]
Comment (withelm) [reply LgymApi.Api/Features/Trainer/Controllers/TrainerRelationshipController.cs]: Done in 14c4fb7: added ProducesResponseType(Status400BadRequest) on this endpoint.

File: LgymApi.Api/Features/Trainer/Controllers/TrainerRelationshipController.cs [hunk 2]
```diff
@@ -73,6 +80,87 @@ public async Task<IActionResult> GetDashboardTrainees([FromQuery] TrainerDashboa
...
+
+        if (!Guid.TryParse(request.ExerciseId, out var parsedExerciseId))
+        {
+            throw AppException.BadRequest(Messages.ExerciseIdRequired);
+        }
+
+        var trainer = HttpContext.GetCurrentUser();
+        var result = await _trainerRelationshipService.GetTraineeExerciseScoresChartDataAsync(trainer!, parsedTraineeId, parsedExerciseId);
+        return Ok(_mapper.MapList<ExerciseScoresChartData, ExerciseScoresChartDataDto>(result));
+    }
```
[the remark this answers is not in the document]
Comment (withelm) [reply LgymApi.Api/Features/Trainer/Controllers/TrainerRelationshipController.cs]: Done in 14c4fb7: added ProducesResponseType(Status400BadRequest) on this endpoint.

File: LgymApi.Api/Features/Trainer/Controllers/TrainerRelationshipController.cs [hunk 3]
```diff
@@ -73,6 +80,87 @@ public async Task<IActionResult> GetDashboardTrainees([FromQuery] TrainerDashboa
...
+    {
+        if (!Guid.TryParse(traineeId, out var parsedTraineeId))
+        {
+            throw AppException.BadRequest(Messages.UserIdRequired);
+        }
+
+        var trainer = HttpContext.GetCurrentUser();
+        var dates = await _trainerRelationshipService.GetTraineeTrainingDatesAsync(trainer!, parsedTraineeId);
+        return Ok(dates);
+    }
```
[the remark this answers is not in the document]
Comment (withelm) [reply LgymApi.Api/Features/Trainer/Controllers/TrainerRelationshipController.cs]: Done in 14c4fb7: added ProducesResponseType(Status400BadRequest) on this endpoint.

File: LgymApi.Api/Features/Trainer/Controllers/TrainerRelationshipController.cs [hunk 4]
```diff
@@ -73,6 +80,87 @@ public async Task<IActionResult> GetDashboardTrainees([FromQuery] TrainerDashboa
...
+    {
+        if (!Guid.TryParse(traineeId, out var parsedTraineeId))
+        {
+            throw AppException.BadRequest(Messages.UserIdRequired);
+        }
+
+        var trainer = HttpContext.GetCurrentUser();
+        var result = await _trainerRelationshipService.GetTraineeEloChartAsync(trainer!, parsedTraineeId);
+        return Ok(_mapper.MapList<EloRegistryChartEntry, EloRegistryBaseChartDto>(result));
+    }
```
[the remark this answers is not in the document]
Comment (withelm) [reply LgymApi.Api/Features/Trainer/Controllers/TrainerRelationshipController.cs]: Done in 14c4fb7: added ProducesResponseType(Status400BadRequest) on this endpoint.

File: LgymApi.Api/Features/Trainer/Controllers/TrainerRelationshipController.cs [hunk 5]
```diff
@@ -73,6 +80,87 @@ public async Task<IActionResult> GetDashboardTrainees([FromQuery] TrainerDashboa
...
+    {
+        if (!Guid.TryParse(traineeId, out var parsedTraineeId))
+        {
+            throw AppException.BadRequest(Messages.UserIdRequired);
+        }
+
+        var trainer = HttpContext.GetCurrentUser();
+        var result = await _trainerRelationshipService.GetTraineeTrainingByDateAsync(trainer!, parsedTraineeId, request.CreatedAt);
+        return Ok(_mapper.MapList<TrainingByDateDetails, TrainingByDateDetailsDto>(result));
+    }
```
[the remark this answers is not in the document]
Comment (withelm) [reply LgymApi.Api/Features/Trainer/Controllers/TrainerRelationshipController.cs]: Done in 14c4fb7: added ProducesResponseType(Status400BadRequest) on this endpoint.
